### ISSUE TITLE
Issues with PowerMock

### ISSUE THREAD

**Issue by pitbulk** [onelogin/java-saml#52]
This toolkit uses PowerMock in order to test its methods.

There are some problems with PowerMock described in the [documentation](https://github.com/onelogin/java-saml/tree/v2.0.0#unit-tests).

We should find a way to fix those problems. Right now it seems is not possible to execute test on Java8 (Java7 requires a extra command to disable the bytecode verification.

**Comment by luuuis** [comment]
Are you open to removing PowerMock completely? I have found that it is not necessary to use byte code manipulation when the code being tested is designed with testing in mind.

For `java-saml` this might mean introducing some interfaces instead of using classes directly.

**Comment by pitbulk** [comment]
If you review the unit test you will see that I added PowerMock in order to be able to simulate HTTP Requests. How is your proposal to manage it?

Im open to suggestions, I dont consider me as a Java expert.

This java-saml toolkit is a port of the PHP/Python toolkits and I tried to implement as similar as possible in order to help me with the global maintenance so when I include a new feature in one I port the functionality on the rest.
